TELCODOCS-646 PR for configure power savings states (#56865)
openshift/openshift-docs: modules/ztp-using-pgt-to-configure-power-states.adoc

// Module included in the following assemblies:
//
// * scalability_and_performance/ztp_far_edge/ztp-advanced-policy-config.adoc

:_module-type: CONCEPT

[id="ztp-using-pgt-to-configure-power-saving-states_{context}"]
= Configuring power states using PolicyGenTemplates CRs

> For low latency and high-performance edge deployments, it is necessary to disable C-states and P-states.

Reviewer: ```suggestion
For low latency and high-performance edge deployments, it is necessary to disable or limit C-states and P-states.
```

> When C-states and P-states are disabled, the CPU runs at a constant frequency, which is typically the maximum turbo frequency. This ensures that the CPU is always running at its maximum speed, which results in high performance and low latency.

Reviewer: ```suggestion
With this configuration, the CPU runs at a constant frequency, which is typically the maximum turbo frequency. This ensures that the CPU is always running at its maximum speed, which results in high performance and low latency.
```

This leads to the best latency for workloads.

> However, this also leads to high power consumption, which might not be necessary for all workloads.

Reviewer: ```suggestion
However, this also leads to the highest power consumption, which might not be necessary for all workloads.
```

> Workloads can be classified as critical or non-critical, with critical workloads requiring disabled C-state and P-state settings for high performance and low latency, while non-critical workloads use C-state and P-state settings for power savings at the expense of some latency and performance. You can configure the following three power states using zero touch provisioning (ZTP):

Reviewer: ```suggestion
Workloads can be classified as critical or non-critical, with critical workloads requiring disabled or limited C-state and P-state settings for high performance and low latency, while non-critical workloads use C-state and P-state settings for power savings at the expense of some latency and performance. You can configure the following three power states using zero touch provisioning (ZTP):
```

* High-performance mode provides ultra low latency at the highest power consumption.
* Performance mode provides low latency at a relatively high power consumption.
* Power saving balances reduced power consumption with increased latency.

The baseline `PerformanceProfile` CR (`PerformanceProfile.yaml`) ships with the power state set to `performance` mode as shown in the following example:

[source,yaml]
----
apiVersion: performance.openshift.io/v2
kind: PerformanceProfile
metadata:
  [...]
spec:
  [...]
  workloadHints: <1>
    realTime: true
    highPowerConsumption: false
    perPodPowerManagement: false
----

> <1> The `workloadHints` field defines the set of upper level flags for different type of workloads. This default configuration is for a low latency, performance mode.

Reviewer: Remove these lines and replace with just "The default configuration is for a low latency, performance mode."
